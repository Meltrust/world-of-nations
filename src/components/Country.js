import React from 'react';
import PropTypes from 'prop-types';

const Country = (props) => {
  const { countrySpecs } = props;
  const {
    id,
    name,
  } = countrySpecs;

  return (
    <div
      id={id}
      className="border border-white light-pink text-center collection-item d-flex flex-column justify-content-center"
    >
      <h1>
        {name}
      </h1>
      <span className="align-bottom">
        <p className="m-0">
          Population: 300 million

        </p>
      </span>
    </div>
  );
};

Country.propTypes = {
  countrySpecs: PropTypes.element.isRequired,
  name: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
};

export default Country;
