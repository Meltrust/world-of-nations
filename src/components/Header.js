import React from 'react';

import { FaAngleLeft, FaMicrophone, FaCog } from 'react-icons/fa';
import { Link } from 'react-router-dom';

const Header = () => (
  <nav className="navbar fixed-top strong-pink ">
    <Link to="/" className="text-white">
      <FaAngleLeft className="ms-2 me-2 home-arrow-button" />
    </Link>
    <h1 className="col-9 fs-4 text-white">World of Nations</h1>
    <FaMicrophone className="mx-0" />
    <FaCog className="ms-2 me-3 text-white" />
  </nav>
);

export default Header;
