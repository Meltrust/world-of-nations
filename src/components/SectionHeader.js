import React from 'react';
import PropTypes from 'prop-types';

const SectionHeader = (props) => {
  const { name, population } = props;
  return (
    <div className="section-header col-12 d-flex flex-column justify-content-center align-items-end">
      <h1 className="fw-bold text-white text-center m-0 p-3 px-0  pb-0 section-header-text">{name}</h1>
      <div className="d-flex flex-column">
        <small className="text-white align-self-center text-center p-0 m-0">
          population:

          { `    ${population}` }
        </small>
      </div>
    </div>
  );
};

SectionHeader.propTypes = {
  name: PropTypes.string.isRequired,
  population: PropTypes.number.isRequired,
};

export default SectionHeader;
