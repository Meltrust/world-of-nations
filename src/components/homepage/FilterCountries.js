import React from 'react';

const FilterCountries = () => (
  <div className="row">
    <form>
      <input
        className="form-control m-2"
        placeholder="Start typing country name..."
        onChange=""
      />
    </form>
  </div>
);
export default FilterCountries;
