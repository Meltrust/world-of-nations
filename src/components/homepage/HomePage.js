import React, { useEffect } from 'react';
import { Outlet, Link } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import SectionHeader from '../SectionHeader';

import Country from '../Country';
import { fetchCountries } from '../../redux/countries/countries';
import FilterCountries from './FilterCountries';

const HomePage = () => {
  const dispatch = useDispatch();
  const countriesData = useSelector((state) => state.countriesReducer);

  useEffect(() => {
    if (countriesData.countriesDisplay.length === 0) {
      console.log('useeffect');
      dispatch(fetchCountries());
    }
  }, []);

  console.log(countriesData.countriesDisplay[0]);
  if (countriesData.loading) {
    return <h2 className="text-white">Loading...</h2>;
  }
  if (countriesData.error) {
    return <h2 className="text-white">{countriesData.error}</h2>;
  }

  return (
    <div className="col-12 row m-0 mt-5">
      <SectionHeader title="Whole World" />
      <FilterCountries />
      {countriesData // conditional
           && countriesData.countriesDisplay // conditional
               && countriesData.countriesDisplay.map(
                 (country) => (
                   <Link
                     to={`/target/${country.id}`}
                     key={country.name}
                     className="p-0 col-6 col-md-4 col-lg-3 col-xl-2 list-links "
                   >
                     <Country
                       key={country.id}
                       id={country.id}
                       name={country.name}
                       population={country.population}
                     />
                   </Link>
                 ),
               )}
      <Outlet />
    </div>
  );
};

export default HomePage;
