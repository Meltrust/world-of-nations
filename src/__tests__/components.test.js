import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Header from '../components/Header';
import store from '../redux/configureStore';
import HomePage from '../components/homepage/HomePage';
import FilterCountries from '../components/homepage/FilterCountries';
import SectionHeader from '../components/SectionHeader';
import Country from '../components/Country';

describe('Components snapshot testing', () => {
  test('Home page testing', () => {
    const homePage = renderer.create(
      <Provider store={store}>
        <HomePage />
      </Provider>,
    ).toJSON;
    expect(homePage).toMatchSnapshot();
  });
  test('Renders search box', () => {
    const box = renderer.create(
      <Provider store={store}>
        <FilterCountries />
      </Provider>,
    ).toJSON;
    expect(box).toMatchSnapshot();
  });
  test('Renders Section Header', () => {
    const sectionHead = renderer.create(
      <Provider store={store}>
        <SectionHeader />
      </Provider>,
    ).toJSON;
    expect(sectionHead).toMatchSnapshot();
  });
  test('Renders Countries', () => {
    const test = { id: 'SDN', name: 'Sudan' };

    const country = renderer.create(
      <Provider store={store}>
        <Country collectionInfo={test} />
      </Provider>,
    ).toJSON;
    expect(country).toMatchSnapshot();
  });

  test('Header rendering', () => {
    const header = renderer.create(
      <Provider store={store}>
        <BrowserRouter><Header /></BrowserRouter>
      </Provider>,
    ).toJSON;
    expect(header).toMatchSnapshot();
  });
});
