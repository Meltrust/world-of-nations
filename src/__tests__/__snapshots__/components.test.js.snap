// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components snapshot testing Header rendering 1`] = `[Function]`;

exports[`Components snapshot testing Home page testing 1`] = `[Function]`;

exports[`Components snapshot testing Renders Countries 1`] = `[Function]`;

exports[`Components snapshot testing Renders Section Header 1`] = `[Function]`;

exports[`Components snapshot testing Renders search box 1`] = `[Function]`;
